catalog/resource_catalog_test.go

```diff
@@ -343,6 +343,63 @@ func TestCatalogCreateDeltaSharing(t *testing.T) {
 	}.ApplyNoError(t)
 }
 
+func TestCatalogCreateForeign(t *testing.T) {
+	qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "POST",
+				Resource: "/api/2.1/unity-catalog/catalogs",
+				ExpectedRequest: catalog.CatalogInfo{
+					Name:    "a",
+					Comment: "b",
+					Options: map[string]string{
+						"a": "b",
+					},
+					ConnectionName: "foo",
+				},
+				Response: catalog.CatalogInfo{
+					Name:    "a",
+					Comment: "b",
+					Options: map[string]string{
+						"a": "b",
+					},
+					ConnectionName: "foo",
+					MetastoreId:    "e",
+					Owner:          "f",
+				},
+			},
+			{
+				Method:   "DELETE",
+				Resource: "/api/2.1/unity-catalog/schemas/a.default?",
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/unity-catalog/catalogs/a?",
+				Response: catalog.CatalogInfo{
+					Name:    "a",
+					Comment: "b",
+					Options: map[string]string{
+						"a": "b",
+					},
+					ConnectionName: "foo",
+					MetastoreId:    "e",
+					Owner:          "f",
+				},
+			},
+		},
+		Resource: ResourceCatalog(),
+		Create:   true,
+		HCL: `
+		name = "a"
+		comment = "b"
+		options = {
```

Comment:
@tanmay-db I am rather unfamiliar with either Go or terraform, what I could not understand is the test result doesn't include the `options` field but I can't figure out why it's the case.